Add yes/no question to the processing

lesics opened the pull request.

nvbln committed:
change library name to be more explicit

Co-Authored-By: lesics <40127153+lesics@users.noreply.github.com>

nvbln committed:
Change library name to be more explicit

Co-Authored-By: lesics <40127153+lesics@users.noreply.github.com>

nvbln committed:
Also add examples to the question analysis

nvbln committed:
Make string comparison case-insensitive

nvbln committed:
Add yes/no keyword extraction

Made by: @lesics

nvbln committed:
Move dict of difficult words to wQ

Made by: @lesics

nvbln committed:
Make yes/no query

Made by: @lesics

nvbln committed:
Deal with difficult property 'play [on instrument]'

Made by: @lesics

nvbln commented on syntacticAnalysis.py:
```diff
@@ -81,6 +75,17 @@ def syntacticAnalysis(nlp, line):
         if attr_pos == 0:
             keywords.append((getPhrase(question, attr_pos), "question_word"))
 
+    elif root_pos == 0 or aux_pos == 0:
```
Aren't these requirements a bit too loose? Or is that the reason you put them at the end?

lesics replied:
Yeah, I still need to do more checking on these questions

nvbln replied:
That's not going to be part of the PR?

nvbln commented on syntacticAnalysis.py:
```diff
@@ -81,6 +75,17 @@ def syntacticAnalysis(nlp, line):
+        if aux_pos == 0:
+            keywords.append((getPhrase(question, aux_pos), "question_word"))
+            keywords.append((getPhrase(question, root_pos), "property"))
+            keywords.append((getPhrase(question, pobj_pos), "property_attribute"))
```
Can you clarify what a property_attribute is?

lesics replied:
It's an attribute that has to exist for the given property - Did X do Y with Z? Is Z Y of X? Z here is  the property attribute, Y - the attribute

nvbln replied:
Could you maybe add this in comments?

nvbln commented on wikidataQuery.py:
```diff
@@ -1,16 +1,50 @@
-            property_id = searchEntity(keyword[0], "property")
+        if keyword[1] == "question_word":
+            if keyword[0] in w_words_list:
+                #TODO update property & possibly query based on that
```
\+ Add a list and count type.

nvbln commented on wikidataQuery.py:
```diff
@@ -1,16 +1,50 @@
             entity_id = searchEntity(keyword[0], "entity")
+            
+        elif keyword[1] == "property_attribute":
+            # TODO attribute is not always entity, right? needs to be fixed
```
No, what would make you reach that conclusion? I do not see the problem.

lesics replied:
Hm, maybe you're right, I'm not sure yet. Got confused with the Is Michael Jackson dead? question

nvbln commented on wikidataQuery.py:
```diff
@@ -1,16 +1,50 @@
+            # TODO attribute is not always entity, right? needs to be fixed
+            prop_attribute_id = searchEntity(keyword[0], "entity")
+            
+    answer = None
```
Alternatively, can't you simply use `return submitQuery()` and `return submitCheckQuery()`? I think that works out fine in python, even if neither if-statement is entered.

lesics replied:
I prefer one return, it's much more elegant

nvbln replied:
I prefer multiple returns, it's more elegant. ;)

nvbln commented on wikidataQuery.py:
```diff
@@ -23,6 +57,8 @@ def searchEntity(entity, string_type):
     params['search'] = entity.rstrip()
     json = requests.get(url,params).json()
     
+    print(entity, '->', json['search'][0]['label'])
```
Is this for debugging purposes? If so, can you please remove it?

lesics replied:
```suggestion
    #print(entity, '->', json['search'][0]['label'])
```

nvbln replied:
```suggestion
```

Since, if you're done with the feature, there would be no need for debugging statements anymore right?

Although we could decide on setting some debugging flag that prints all debugging messages. Might be something to discuss later.

nvbln replied:
Assuming the verbose branch gets merged first, you can now change this to:
```
if settings.verbose:
    print(entity, '->', json['search'][0]['label'])
```

nvbln commented on wikidataQuery.py:
```diff
@@ -45,3 +81,24 @@ def submitQuery(entity_id, property_id):
+    data = requests.get(url, params={'query': query, 'format': 'json'}).json()
+
+    answer = []
+    if data['boolean'] == True:
```
`if data['boolean'] == True:` can just be `if data['boolean']:` right?

lesics replied:
I prefer to be explicit

nvbln reviewed (changes requested):
Changes are not necessarily necessary. Depends on the answers on my comments.

nvbln committed:
Replace the placeholder code with actual answers

nvbln committed:
Track the number of questions checked

nvbln committed:
Add verbose functionality

nvbln committed:
Use verbose in the wrapper

nvbln committed:
Add (example) verbose statements

nvbln committed:
Make evaluateQuestions more verbose

nvbln committed:
evaluateTestQuestions index out of range hotfix

nvbln committed:
Allow for the -t and -v combination

nvbln committed:
Only display loading message in verbose

nvbln committed:
Introduce local verbosity when combined with test

nvbln commented on wikidataQuery.py:
```diff
@@ -45,3 +81,24 @@ def submitQuery(entity_id, property_id):
+    else:
+        answer = ['No']
+    
+return answer
```
Did you not get errors while running this file? Because the last return statement is outside the function body.

nvbln committed:
Do not count incorrect answers as correct

nvbln committed:
Fix 'incorrect' string comparisons due to spaces

nvbln committed:
Add verbosity to the test

nvbln committed:
Fix out of index error while running --test

nvbln committed:
Print predicted question structure

nvbln committed:
Add new question type

A category for questions like: "From which album is Dancing Queen?"

nvbln committed:
Add further specification to entity for a question

nvbln committed:
Sharpen the X of Y category requirements

nvbln committed:
Loosen requirements for X's Y questions

Also add everything behind a case as property
if there is no second attribute.

nvbln committed:
Change X is Y question type requirements

'Det' is used as a requirement because SpaCy translates
words like 'which' and 'what' to the 'Det' type
in these kind of sentences.

nvbln committed:
Add question type

Question type: 'X was Y of Z'

nvbln committed:
Add the root tag in X [verb] Y questions

Commit:
Put print statement in verbose

Commit:
IMplement feedback

lesics committed:
Merge branch 'master' into yes_no
